### ISSUE TITLE
[Bug] Ward with multiple Desks staff return issue

### ISSUE THREAD

**Issue by ageekhere** [CorsixTH/CorsixTH#2429]
### Describe the issue

Issues with multiple Desks and nurses for the Ward.
If you do have 2 Desks with 2 nurses in the Ward and one goes to the staff or another room they will not return to the Ward if a nurse is already in there because they are not needed so they wait in the hall.

### Steps to Reproduce

1. Create a GP, Ward with multiple Desks with nurses and a staff room

2. Wait for nurses to go to the staff room

3. Check how many return to the Ward to work and how many will wait in the hall

### Expected Behaviour

Nurses should go to Wards that have empty desks in them if all rooms that need nurses have a nurse allocated to them rather then waiting in the halls.

Because of this bug you cannot use Wards with multiple Desks because you have to manually put nurses back in there.

### Save Game

Save game attached to report: No

### CorsixTH Version

0.67

### Operating System

Windows 11

### Theme Hospital Version

GOG.com

**Comment by MandrakeHorse** [comment]
I haven't been able to reproduce this bug. Instead of wandering the corridors, all of my nurses return to the Ward after resting in the Staff Room. Can you upload a save file (or video footage) demonstrating this unusual behaviour?

My save files with unsuccessful attempts at reproducing this issue: [nursesWard_issue2429.zip](https://github.com/CorsixTH/CorsixTH/files/12775708/nursesWard_issue2429.zip)

**Comment by ageekhere** [comment]
Ok I looked into this issue a bit more. 
Have allow staff to leave rooms ticked. 

If the nurses are never called to another room then they will return to the Ward. However this is where the issue can happen.
You have 2 nurses in the Ward and 1 in the Pharmacy, the nurse in the Pharmacy leaves and goes to the staff room. The ward is empty so a nurse leaves the Ward and goes to the Pharmacy. Now that the nurse has left the Warn it will never get 2 nurses again and the returning nurse that was in the Pharmacy will walk the halls. 

So in a game even if you put 5 nurses in the Ward with desks at some point if they leave to go to another room they will never go back to the Ward and you will always be left with 1 nurse in there.
 
The fix would be for idle nurses is first check if there are unallocated desks in the Ward, if there are send the nurse there. If not then the nurse can be idle in the hall.

**Comment by Alberth289346** [comment]
In the cases of nurses having to go elsewhere, you'll want to have similar considerations but in reverse. First pick idling nurses, then pick a nurse in a ward with more than one nurse (assuming that each nurse contributes an equal amount of healing), and then pick a nurse in an otherwise unused room.

This applies in particular when you have multiple wards.

**Comment by ageekhere** [comment]
yes this would be good, atm having multiple Desks in the Ward is too hard to manage in a big hospital as you have to keep moving nurses back in there.

**Comment by lewri** [comment]
Generally speaking, staff won't seek out a room that doesn't currently require staff.
But I get the point. If a Nurse is in the ward, there are patients present, and there's a free desk -- the nurse could go occupy the desk. Chances are the logic for finding work are a bit flawed in that regard.
